controllers/spacerequest/spacerequest_controller.go

```diff
@@ -486,7 +487,18 @@ func (r *Reconciler) ensureSecretForProvisionedNamespaces(ctx context.Context, m
 
 		case len(secretList.Items) > 1:
 			// some unexpected issue causing to many secrets
-			return fmt.Errorf("invalid number of secrets found. actual %d, expected %d", len(secretList.Items), 1)
+			// this ca be caused by the client cache which is not up-to-date immediately after the first secret is created.
+			logger.Error(fmt.Errorf("invalid number of secrets found. actual %d, expected %d", len(secretList.Items), 1), "defaulting on the first secret in alphabetical order")
```

Comment:
should we use log.Info here? something like the following

```suggestion
			logger.Info("invalid number of secrets found: defaulting on the first secret in alphabetical order", "actual", len(secretList.Items), "expected", 1)
```

Reply:
it's fine to use `Info` as well indeed. I haven't thought about the stacktrace. The main reason to use `Error` was that it is an error case actually :) , and I don't expect this to happen so often, but grepping the string should be easy as well. 

Thanks for the suggestion.